src/AddressBook.java

```diff
@@ -64,7 +68,7 @@ public Person get(int index) {
      * Clears this address book persons list and updates the tables.
      */
     public void clear() {
-        if (persons == null || persons.isEmpty()) {
```

Comment:
That's weird. I didn't notice that bit of useless code. I was checking everything too when I added comments.

Reply:
Yeah so is it ok if we removed persons==null?
I cant find way to hit that branch and I don't see why it would ever happen

Reply:
yeah, seems to me that it shouldn't ever be right.